plugins/GSdx/GSUtil.cpp

```diff
@@ -396,6 +396,41 @@ D3D_FEATURE_LEVEL GSUtil::CheckDirect3D11Level(IDXGIAdapter *adapter, D3D_DRIVER
 	return SUCCEEDED(hr) ? level : (D3D_FEATURE_LEVEL)0;
 }
 
+GSRendererType GSUtil::GetBestRenderer()
+{
+	CComPtr<IDXGIAdapter1> adapter;
+	DXGI_ADAPTER_DESC1 desc;
+	CComPtr<IDXGIFactory1> dxgi_factory;
+	GSRendererType BestRenderer = GSRendererType::Default;
+	bool Result = false;
+
+	if (GSUtil::CheckDXGI())
+	{
+		Result = SUCCEEDED(CreateDXGIFactory1(IID_PPV_ARGS(&dxgi_factory)));
+	}
+
+	if(GSUtil::CheckD3D11() && Result)
```

Comment:
The `if` is inconsistent with the other `if` statements you used.

A suggestion: maybe use early return instead? Something like:
```
if (!GSUtil::CheckDXGI() || !GSUtil::CheckD3D11())
	return GSRendererType::DX9_HW;

if (FAILED(CreateDXGIFactory1(IID_PPV_ARGS(&dxgi_factory))))
	return GSRendererType::DX9_HW;

// and so on.
```

Side note: I'm not sure CheckDXGI and CheckD3D11 are necessary now that we're not supporting XP - I think Vista SP2 has both dlls, but I'd need to confirm it first.